pkg/scheduler/predicates.go

```diff
@@ -169,6 +169,125 @@ func PodFitsHost(pod api.Pod, existingPods []api.Pod, node string) (bool, error)
 	return pod.Spec.Host == node, nil
 }
 
+type NodeLabelChecker struct {
+	info     NodeInfo
+	labels   []string
+	presence bool
+}
+
+func NewNodeLabelPredicate(info NodeInfo, labels []string, presence bool) FitPredicate {
+	labelChecker := &NodeLabelChecker{
+		info:     info,
+		labels:   labels,
+		presence: presence,
+	}
+	return labelChecker.CheckNodeLabelPresence
+}
+
+// CheckNodeLabelPresence checks whether a particular label exists on a minion or not, regardless of its value
```

Comment:
This doesn't seem to be checking about a "particular" label, since you are looping through n.labels. My understanding of this function was more like
- If "presence" is false, then return false if any of the requested labels matches any of the minion's labels, otherwise return true
- If "presence" is true, then return false if any of the requested labels does not match any of the minion's labels, otherwise return true
If I'm understanding correctly, then please update the comment.